app/views/vendors/show.html.erb

```diff
@@ -0,0 +1,56 @@
+<h1> Welcome, <%=@vendor.name%>! </h1>
+
+
+<h2>Sales: </h2> <br>
+<%= button_to "Add a sale", new_sale_path(@vendor.id), method: :get %>
+  <% total_sales = 0 %>
+  <% @vendor.sales.each do |sale| %>
+    <ul>
+
+      <li> Amount: $ <%= number_with_precision (sale.amount/100.0), precision: 2 %> </li>
+      <li> Product: <%= Product.find(sale.product_id).name %> </li>
+      <li> Purchase time:
+      <% if !sale.purchase_time.nil? %>
+         <%= sale.purchase_time.strftime("%m/%d/%Y at %I:%M%p") %> </li>
+        <% end %>
+      <%= button_to "Delete sale", sale_path(sale.id), method: :delete, data: {confirm: "Are you sure?"} %>
+    </ul>
+  <% total_sales += sale.amount %>
+  <% end %>
```

Comment:
Try to keep as much calculation out of the view as possible. Total sales could be calculated on the model and passed through the controller to the view.